rs/types/management_canister_types/src/lib.rs

```diff
@@ -1073,6 +1114,38 @@ impl CanisterStatusResultV2 {
         NumBytes::from(self.memory_size.0.to_u64().unwrap())
     }
 
+    pub fn wasm_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.heap_size.0.to_u64().unwrap())
+    }
+
+    pub fn stable_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.stable_size.0.to_u64().unwrap())
+    }
+
+    pub fn global_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.global_size.0.to_u64().unwrap())
+    }
+
+    pub fn wasm_binary_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.wasm_binary_size.0.to_u64().unwrap())
+    }
+
+    pub fn custom_sections_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.custom_sections_size.0.to_u64().unwrap())
+    }
+
+    pub fn canister_history_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.canister_history_size.0.to_u64().unwrap())
+    }
+    
+    pub fn wasm_chunk_store_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.wasm_chunk_store_size.0.to_u64().unwrap())
+    }
+
+    pub fn snapshot_memory_size(&self) -> NumBytes {
+        NumBytes::from(self.memory_metrics.snapshot_size.0.to_u64().unwrap())
+    }
```

Comment:
Are these functions truly needed? From what I can tell we're not using them. Do you expect some use case?

Reply:
This is because the fields of `CanisterStatusResultV2` are not public.

Reply:
But where are they used? The values are retrieved from `CanisterState` and we populate `CanisterStatusResultV2` but we never need to read the values from `CanisterStatusResultV2` in the replica code, am I missing something?

If this is to allow canisters to extract the values, then I'd leave the addition to the public library that canisters are supposed to depend on when dealing with management canister types. The versions here are internally used in the replica and so I would only add what's needed for the replica to function correctly.

Reply:
Ah, I just realized this is needed for the tests. Ok, then I suppose we can leave it as is. I believe there's a reason we don't make the fields public to ensure that you can only create a `CanisterStatusResultV2` through the constructor.